### ISSUE TITLE
500 (Internal Server Error) in some cases after restarting LND & RTL

### ISSUE THREAD

**Issue by andyhowell8** [Ride-The-Lightning/RTL#245]
**Describe the bug**
Server 500 on the following routes `/peers` & `/channels`
 
UI never retrieves channel or peer info, unless RTL is restarted

```
{"message":"Fetching All Channels Failed!","error":{"error":"pubkey string is empty","message":"pubkey string is empty","code":2}}
```

```
{"message":"Peers Fetched Failed!","error":{"error":"need: lnc pubkeyhash@hostname","message":"need: lnc pubkeyhash@hostname","code":2}}
```

**To Reproduce**
1. Open RTL
2. Restart LND and do not restart RTL
3. Refresh RTL user interface
4. Login to RTL, Unlock LND wallet
5. Open browser network tools and observe server 500 on the API routes `/peers` & `/channels`, other routes work as expected 

**Expected behavior**

**Actual behavior**
Tell us what happens instead

**Screenshots**

**Your environment**
* version of `RTL` = `0.6.4`
* version of `lnd` = `0.9.0`
* version of `bitcoind` = `0.19.0`
* any other relevant environment details

**Additional context**
Restart RTL fixes error with `/peers` & `/channels`
Using RTL docker container

**Comment by ShahanaFarooqui** [comment]
@andyhowell8 These errors appear when LND has recently been unlocked/initialized. RTL has 90 seconds wait time for lnd to sync after unlock. With older LND versions, it used to sync within 90 seconds but now, it seems to take more time than that. So, if the LND is not synced and RTL queries for peers & channels, it receives errors and shows a red border around related information cards.

But, it does not require RTL server restart. Just a browser refresh, after LND is sync, should be able to fetch the information.

**Comment by andyhowell8** [comment]
Ok I will test again tomorrow, I'm pretty sure I left RTL for a long time and it did not resolve the issue but I will confirm tomorrow. 

I do see the red box and the only thing that resolved it was an RTL restart. 

Let me check if LND is still syncing and will report back, thanks

**Comment by andyhowell8** [comment]
Ok so I have retested this issues, 
I made sure LND was synced after a restart, and the issue remains

The only thing that fixed the server 500 was a restart or RTL, as soon as I logged in after the restart get the correct response for `/peers` & `/channels` work as expected


LND log file
```
2020-01-31 10:52:22.973 [ERR] RPCS: [/lnrpc.Lightning/ConnectPeer]: need: lnc pubkeyhash@hostname
2020-01-31 10:52:22.983 [ERR] RPCS: [/lnrpc.Lightning/OpenChannelSync]: pubkey string is empty
2020-01-31 10:52:49.077 [ERR] RPCS: [/lnrpc.Lightning/ConnectPeer]: need: lnc pubkeyhash@hostname
2020-01-31 10:52:49.107 [ERR] RPCS: [/lnrpc.Lightning/OpenChannelSync]: pubkey string is empty
```

**Comment by ShahanaFarooqui** [comment]
Thanks, let us try to reproduce it at our end. Till now, we always get response only by F5/browser refresh after LND sync.

**Comment by saubyk** [comment]
@andyhowell8 
Can you please confirm if refreshing the browser addresses the issue?
or
You have to restart the RTL server?

**Comment by saubyk** [comment]
@andyhowell8 You can ignore the above, we are able to reproduce this issue. Thanks.

**Comment by ShahanaFarooqui** [comment]
Fixed and pushed to feature/0.6.5. The fix will be release with version 0.6.5.

**Comment by andyhowell8** [comment]
Fantastic, looking forward to testing

**Comment by downet** [comment]
Hi, I am getting the same error in BTCPayServer RTL (v0.11.1-beta).

`
[2021-09-08T20:12:25.973Z] ERROR: Peers => Connect Peer Error: {"name":"StatusCodeError","statusCode":500,"message":"500 - {\"error\":\"need: lnc pubkeyhash@hostname\",\"code\":2,\"message\":\"need: lnc pubkeyhash@hostname\",\"details\":[]}","error":{"error":"need: lnc pubkeyhash@hostname","code":2,"message":"need: lnc pubkeyhash@hostname","details":[]},"options":{"url":"http://lnd_bitcoin:8080/v1/peers","rejectUnauthorized":false,"json":true,"headers":{},"method":"POST","qs":{},"form":"{\"perm\":false}","simple":true,"resolveWithFullResponse":false,"transform2xxOnly":false},"response":{"statusCode":500,"body":{"error":"need: lnc pubkeyhash@hostname","code":2,"message":"need: lnc pubkeyhash@hostname","details":[]},"headers":{"content-type":"application/json","trailer":"Grpc-Trailer-Content-Type","date":"Wed, 08 Sep 2021 20:12:25 GMT","connection":"close","transfer-encoding":"chunked"},"request":{"uri":{"protocol":"http:","slashes":true,"auth":null,"host":"lnd_bitcoin:8080","port":"8080","hostname":"lnd_bitcoin","hash":null,"search":null,"query":null,"pathname":"/v1/peers","path":"/v1/peers","href":"http://lnd_bitcoin:8080/v1/peers"},"method":"POST","headers":{"content-type":"application/x-www-form-urlencoded","accept":"application/json","content-length":14}}}}
`

**Comment by ShahanaFarooqui** [comment]
@downet 

This bug fix was released with hotfix release 0.11.2. Please update the RTL version to fix the issue

**Comment by downet** [comment]
@ShahanaFarooqui thanks! Tested and working, going to create PR for BTCPayServer
